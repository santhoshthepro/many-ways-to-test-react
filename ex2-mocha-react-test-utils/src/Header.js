var React = require('react');

var Header = React.createClass({
  render: function() {
    var authLinkText = 'Login';
    var authMessage = 'Have an account?';
    if (this.props.user && this.props.user.username) {
      authLinkText = 'Logout';
      authMessage = 'Welcome, ' + this.props.user.username;
    }

    return (
      <nav className="navbar navbar-default">
        <div className="container-fluid">
          <div className="navbar-header">
            <button type="button" className="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
              <span className="sr-only">Toggle navigation</span>
              <span className="icon-bar"></span>
              <span className="icon-bar"></span>
              <span className="icon-bar"></span>
            </button>
            <a className="navbar-brand" href="#">Dogs: The Musical</a>
          </div>

          <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul className="nav navbar-nav">
              <li><a href="#">The Story</a></li>
              <li><a href="#">Cast</a></li>
              <li><a href="#">Director</a></li>
              <li><a href="#">Venue</a></li>
            </ul>
            <form className="navbar-form navbar-right">
              <div className="form-group">
                <input type="text" className="form-control" placeholder="Search" />
              </div>
              <button type="submit" className="btn btn-default">Go</button>
            </form>
            <ul className="nav navbar-nav navbar-right login-block">
              <li className="navbar-text">{authMessage}</li>
              <li><a href="#" className="navbar-link">{authLinkText}</a></li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
});

module.exports = Header;
