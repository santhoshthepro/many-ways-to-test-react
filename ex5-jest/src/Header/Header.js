import React, {PropTypes} from 'react';

const Header = ({ login, logout, user }) => {
  let authLink = login;
  let authLinkText = 'Login';
  let authMessage = 'Have an account?';
  if (user && user.username) {
    authLink = logout;
    authLinkText = 'Logout';
    authMessage = 'Welcome, ' + user.username;
  }

  return (
    <nav className="navbar navbar-default">
      <div className="container-fluid">
        <div className="navbar-header">
          <button type="button" className="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
            <span className="sr-only">Toggle navigation</span>
            <span className="icon-bar"></span>
            <span className="icon-bar"></span>
            <span className="icon-bar"></span>
          </button>
          <a className="navbar-brand" href="#">Dogs: The Musical</a>
        </div>

        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
          <ul className="nav navbar-nav">
            <li><a href="#">The Story</a></li>
            <li><a href="#">Cast</a></li>
            <li><a href="#">Director</a></li>
            <li><a href="#">Venue</a></li>
          </ul>
          <form className="navbar-form navbar-right">
            <div className="form-group">
              <input type="text" className="form-control" placeholder="Search" />
            </div>
            <button type="submit" className="btn btn-default">Go</button>
          </form>
          <ul className="nav navbar-nav navbar-right login-block">
            <li className="navbar-text">{authMessage}</li>
            <li><a href="#" className="navbar-link" onClick={authLink}>{authLinkText}</a></li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

Header.propTypes = {
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  user: PropTypes.object
};

export default Header;
