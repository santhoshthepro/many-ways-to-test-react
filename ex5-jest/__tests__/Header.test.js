import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/Header/Header';

describe('Header', function() {
  let props;

  beforeEach(function() {
    props = {
      user: {
        id: 9,
        username: 'gmtester'
      },
      login: jest.fn(),
      logout: jest.fn()
    };
  });

  test('Header rendered static HTML', () => {
    const component = renderer.create(
      <Header {...props} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
