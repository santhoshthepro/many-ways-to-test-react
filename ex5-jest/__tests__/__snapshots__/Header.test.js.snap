exports[`Header Header rendered static HTML 1`] = `
<nav
  className="navbar navbar-default">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <button
        aria-expanded="false"
        className="navbar-toggle collapsed"
        data-target="#bs-example-navbar-collapse-1"
        data-toggle="collapse"
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
      <a
        className="navbar-brand"
        href="#">
        Dogs: The Musical
      </a>
    </div>
    <div
      className="collapse navbar-collapse"
      id="bs-example-navbar-collapse-1">
      <ul
        className="nav navbar-nav">
        <li>
          <a
            href="#">
            The Story
          </a>
        </li>
        <li>
          <a
            href="#">
            Cast
          </a>
        </li>
        <li>
          <a
            href="#">
            Director
          </a>
        </li>
        <li>
          <a
            href="#">
            Venue
          </a>
        </li>
      </ul>
      <form
        className="navbar-form navbar-right">
        <div
          className="form-group">
          <input
            className="form-control"
            placeholder="Search"
            type="text" />
        </div>
        <button
          className="btn btn-default"
          type="submit">
          Go
        </button>
      </form>
      <ul
        className="nav navbar-nav navbar-right login-block">
        <li
          className="navbar-text">
          Welcome, gmtester
        </li>
        <li>
          <a
            className="navbar-link"
            href="#"
            onClick={[Function]}>
            Logout
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
